nemo_curator/modifiers/slicer.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional, Union
+
+from nemo_curator.modifiers import DocumentModifier
+
+
+class Slicer(DocumentModifier):
+    """
+    Slices a document based on indices or strings.
+    """
+
+    def __init__(
+        self,
+        left: Optional[Union[int, str]] = None,
+        right: Optional[Union[int, str]] = None,
+        include_left: bool = True,
+        include_right: bool = True,
+        strip: bool = True,
+    ):
+        """
+        Args:
+            left (Union[int, str], optional): If the provided value is an int, slice the string from this index (inclusive).
+                If the provided value is a str, slice the string from the first occurence of this substring.
+            right (Union[int, str], optional): If the provided value is an int, slice the string to this index (exclusive).
```

Comment:
can you document the behavior for `None` which is imputed as left=0 and right=len(text)